raster/r.fill.stats/main.c

```diff
@@ -159,12 +159,6 @@ void print_weights_matrix(long int rows, long int cols)
             if (WEIGHTS[i][j] != -1.0) {
                 snprintf(weight_matrix_weight_buf, weight_matrix_line_length,
                          "%06.2f ", WEIGHTS[i][j]);
```

Comment:
```suggestion
                         "%.15g ", WEIGHTS[i][j]);
```

Reply:
`weight_matrix_weight_length` should be `#define `d. Alternatively, use `G_asprintf()` with `char *weight_matrix_line_buf = NULL;` instead of `snprintf()`.

There are numerous other compiler warnings for this module. Are they addressed by your other PRs or do you intend to get rid of compiler warnings for this module?

Reply:
The `print_weights_matrix()` is a "print nice" function with the `-w` argument which outputs nicely (and precisely calculated) in a 80 column row (if it fits). To use `%.15g` turns this upside down:

before (with `%06.2f`):

```
r.fill.stats -m -w input=cfactorbare_1m@PERMANENT output=asdf distance=20 mode=wmean power=2.0 cells=8
W-E size of neighborhood is 5 cells.
S-N size of neighborhood is 5 cells.
Spatial weights neighborhood (cells):
...... 000.00 000.01 000.00 ...... 
000.00 000.14 000.31 000.14 000.00 
000.01 000.31 001.00 000.31 000.01 
000.00 000.14 000.31 000.14 000.00 
...... 000.00 000.01 000.00 ...... 
```

after (with `%.15g`):
```
r.fill.stats -m -w input=cfactorbare_1m@PERMANENT output=asdf distance=20 mode=wmean power=2.0 cells=8
W-E size of neighborhood is 5 cells.
S-N size of neighborhood is 5 cells.
Spatial weights neighborhood (cells):
...... 0 0.0111456180001682 0 ...... 
0 0.135088935932648 0.305572809000084 0.135088935932648 0 
0.0111456180001682 0.305572809000084 1 0.305572809000084 0.0111456180001682 
0 0.135088935932648 0.305572809000084 0.135088935932648 0 
...... 0 0.0111456180001682 0 ...... 
```

I seem to me that this function worked well, there was only an unnecessary part which is removed with this PR.

Reply:
> There are numerous other compiler warnings for this module. Are they addressed by your other PRs or do you intend to get rid of compiler warnings for this module?

They are already dealt with, with #1276.

Reply:
> The `print_weights_matrix()` is a "print nice" function with the `-w` argument which outputs nicely (and precisely calculated) in a 80 column row (if it fits). To use `%.15g` turns this upside down:

OK, please ignore my suggestion.